pages/[user]/[type].tsx

```diff
@@ -233,72 +234,73 @@ export default function Type(props): Type {
   }
 }
 
-export const getServerSideProps: GetServerSideProps = async (context: GetServerSidePropsContext) => {
-  const dateQuery = context.query?.date ?? null;
-  const date = Array.isArray(dateQuery) && dateQuery.length > 0 ? dateQuery.pop() : dateQuery;
+export const getServerSideProps = async (context: GetServerSidePropsContext) => {
+  // get query params and typecast them to string
+  // (would be even better to assert them instead of typecasting)
+  const userParam = context.query.user as string;
+  const typeParam = context.query.type as string;
+  const dateParam = context.query.date as string | undefined;
 
-  const user = await whereAndSelect(
```

Comment:
This function is not type-safe and should be deprecated

Reply:
Is there a way to refactor this function into being typesafe or is that undo-able by design?

Reply:
Not sure - possibly - but I don't see why it'd be worth the hassle of going down a generics rabbit hole for over simply using something that is part of prisma natively

Reply:
In most cases this helper reduced the amount of code significantly and I saw no reason not to wrap native prisma with some decorators on the condition that Prisma could still be accessed directly. But I admit preference of the team did not switch to using the helper function and in most cases ignored the helper altogether, based on this it makes sense to remove.